clcache.py

```diff
@@ -142,7 +218,13 @@ def __init__(self):
             os.makedirs(self.objectsDir)
         lockName = self.cacheDirectory().replace(':', '-').replace('\\', '-')
         timeout_ms = int(os.environ.get('CLCACHE_OBJECT_CACHE_TIMEOUT_MS', 10 * 1000))
-        self.lock = ObjectCacheLock(lockName, timeout_ms)
+
+        cfg = Configuration(self)
```

Comment:
Good catch! There's a hen and egg problem here: the `Configuration` constructor reads the cache configuration. To do so, you first need to acquire a lock. However, to know what kind of lock to acquire you currently need to look at the cache configuration. Oops.

Quite frankly, given that file-based locking doesn't seem to have any downsides so far (at least not timing-wise), I think we could just as well make it the default for the next version and see how it goes.